### ISSUE TITLE
How to train your data

### ISSUE THREAD

**Issue by xiaojidan** [ashafaei/dense-depth-body-parts#3]
Hi, can you also share the training code for your paper, so that we can better understand your idea in your paper?

**Comment by ashafaei** [comment]
I did the training with Caffe, but you can easily use [matconvnet](https://github.com/vlfeat/matconvnet) to do the training as well (See [this example for FCNs](https://github.com/vlfeat/matconvnet-fcn)). The procedure is quite simple and standard.

Take the [UBC3V dataset](https://github.com/ashafaei/ubc3v).
- Train on easy-pose first.
- Then fine-tune on the inter-pose.
- Finally, fine-tune on the hard-pose.

and you'll end up with this model.

**Comment by xiaojidan** [comment]
OK. Thanks @ashafaei

**Comment by xiaojidan** [comment]
Well, do you share me some training codes? I want to know that how you train?

**Comment by Ellenisawake** [comment]
May I know how you input the depth data and label in training? Did you use hdf5 or python layer? 
I’m quite interested in this work and am trying to repeat your training, but got some problem to input the data into caffe. And also I’m curious about how you did the pixel-wise loss calculation?

**Comment by ashafaei** [comment]
I made LMDB databases of pre-processed depth images and the corresponding groundtruths. You can do it with HDF5 or even on-the-fly with the python layers as well. The pixel wise loss calculation is automatically handled with the cross-entropy loss (aka SoftmaxLossLayer). Furthermore, you can specify labels to ignore in the configuration of the SoftmaxLoss, which is useful to mask out the groundtruths. I think it's more helpful if you follow a tutorial for using caffe, or examine some of the available codes such as the recent FCN release (Google FCN Berkeley vision) first. Good luck.

**Comment by Ellenisawake** [comment]
@ashafaei Oh I see. I tried leveldb at first but failed to do so.
You're right. Thank you very much for the advice!
